Environment_Engine/Modify/SetOutlineElements1D.cs

```diff
@@ -33,31 +36,26 @@ public static partial class Modify
         /****               Public Methods              ****/
         /***************************************************/
 
+        [Description("BH.Engine.Environment.Modify.SetOutlineElements1D => Assign a new collection of external 1D boundaries to an Environment Opening")]
+        [Input("opening", "An Environment Opening to update")]
+        [Input("outlineElements1D", "A collection of outline 1D elements to assign to the opening")]
+        [Output("opening", "The updated Environment Opening")]
         public static Opening SetOutlineElements1D(this Opening opening, List<IElement1D> outlineElements1D)
         {
             Opening o = opening.GetShallowClone() as Opening;
-            o.OpeningCurve = new PolyCurve { Curves = outlineElements1D.Cast<ICurve>().ToList() };
+            o.Edges = outlineElements1D.Cast<ICurve>().ToList().ToEdges();
```

Comment:
As long as the edge objects are returned straight away from the query method, you should be able to cast them to edges straight away here. Think all of this will get moved to Analytical_Engine in the long run anyway, but could be good to know.